internal/policies/apparmor/apparmor.go

```diff
@@ -229,6 +230,8 @@ func (m *Manager) applyMachinePolicy(ctx context.Context, e entry.Entry, apparmo
 		// #nosec G204 - We are in control of the arguments
 		cmd := exec.CommandContext(ctx, apparmorParserCmd[0], apparmorParserCmd[1:]...)
 		cmd.Dir = m.apparmorDir
+		smbsafe.WaitExec()
+		defer smbsafe.DoneExec()
```

Comment:
defer encompasses for now the `os.RemoveAll` wich may take some time. I think it’s better to release the lock as soon as possible.

Of course, think about doing it in the error case too :)